ludwig/utils/defaults.py

```diff
@@ -161,16 +166,75 @@ def _upgrade_deprecated_fields(config: Dict[str, Any]):
             )
             feature[TYPE] = NUMBER
 
-    if HYPEROPT in config and "parameters" in config[HYPEROPT]:
-        hparams = config[HYPEROPT]["parameters"]
-        for k, v in list(hparams.items()):
-            substr = "training."
-            if k.startswith(substr):
+    if HYPEROPT in config:
```

Comment:
Thanks for adding this backward-compatibility code!

Could we add a test for `_upgrade_deprecated_fields` or `merge_with_defaults` that asserts on the full upgrade of a legacy config (that uses lots of these code paths)?

Such a test would be an easy source of reference for Ludwig users to see concretely what's changed.

Reply:
Address by PR https://github.com/ludwig-ai/ludwig/pull/2082